adcontrol.php

```diff
@@ -391,6 +391,25 @@ function get_ad_snippet( $section_id, $height, $width, $adblock_unit = 'mrec', $
 		$data_tags = $this->params->cloudflare ? ' data-cfasync="false"' : '';
 		$adblock_ad = $this->get_adblocker_ad( $adblock_unit );
 
+		if ( function_exists( 'is_amp_endpoint' ) && is_amp_endpoint() ) {
+
+      add_action( 'amp_post_template_css', array( $this, 'add_ad_styles' ) );
```

Comment:
The `amp_post_template_css` call should be done in the `insert_adcode()` method to prevent it from being called multiple times. 

Also, it won't work in the new paired/native theme modes, since `amp_post_template_css` is not used. If `current_theme_supports('amp')` you should just print out `<style>` right into the page. So do this when `current_theme_supports('amp')` and do this at `wp_head` action.